auth/src/main/java/com/firebase/ui/auth/ui/accountmanagement/AccountSettingsActivity.java

```diff
@@ -0,0 +1,269 @@
+                        providerName.setText(R.string.idp_name_facebook);
+                        break;
+                    case AuthUI.GOOGLE_PROVIDER:
+                        providerImage.setBackground(
+                                makeColoredCircle(R.color.gdi_white_background));
+                        providerImage.setImageResource(R.drawable.ic_googleg_color_24dp);
+                        providerName.setText(R.string.idp_name_google);
+                        break;
+                    case AuthUI.TWITTER_PROVIDER:
+                        providerImage.setBackground(makeColoredCircle(R.color.gdi_twitter_blue));
+                        providerImage.setImageResource(R.drawable.ic_twitter_bird_white_24dp);
+                        px = pxFromDp(10);
+                        providerImage.setPadding(px, px, px, px);
+                        providerName.setText(R.string.idp_name_twitter);
+                        break;
+                }
+
+            }
+        }
+    }
+
+    private Intent getReauthIntent() {
+        return AuthUI.getInstance().createReauthIntentBuilder()
+                .setTosUrl(mFlowParameters.termsOfServiceUrl)
+                .setLogo(mFlowParameters.logoId)
+                .setProviders(mFlowParameters.providerInfo)
+                .setIsSmartLockEnabled(mFlowParameters.smartLockEnabled)
+                .build();
+    }
+
+    private ShapeDrawable makeColoredCircle(@ColorRes int color) {
+        ShapeDrawable shapeDrawable = new ShapeDrawable();
+        shapeDrawable.setShape(new OvalShape());
+        int resolvedColor = getResources().getColor(color);
+        shapeDrawable.getPaint().setColor(resolvedColor);
+        return shapeDrawable;
+    }
+
+    private int pxFromDp(int dp) {
+        return (int) Math.ceil(dp * getResources().getDisplayMetrics().density);
+    }
+
+    private void maybeShowChangePassword() {
+        View changePasswordLayout = findViewById(R.id.change_password_layout);
+        List<String> providers = mActivityHelper.getCurrentUser().getProviders();
+        if (providers.isEmpty() || providers.contains(AuthUI.EMAIL_PROVIDER)) {
+            changePasswordLayout.setVisibility(View.VISIBLE);
+        } else {
+            changePasswordLayout.setVisibility(View.GONE);
+        }
+    }
+
+    class UnlinkIdpClickListener implements View.OnClickListener {
+        private String mProviderId;
+
+        public UnlinkIdpClickListener(String providerId) {
+            mProviderId = providerId;
+        }
+
+        @Override
+        public void onClick(View v) {
+            mActivityHelper
+                    .getCurrentUser()
+                    .unlink(mProviderId)
+                    .addOnFailureListener(
+                            new TaskFailureLogger(TAG, "Failed to unlink " + mProviderId))
+                    .addOnSuccessListener(new OnSuccessListener<AuthResult>() {
```

Comment:
This OnSuccessListener should be activity scoped (`addOnSuccessListener(activity, listener`).

Might want to go a step further and have this inner class be static and hold a WeakReference to the Activity, which is checks in onClick before launching anything.